common/table.js

```diff
@@ -1527,10 +1527,85 @@
                         columnModels.push(recordCreate.columnToColumnModel(col));
                     });
 
-                    
                     // get the stable facet
                     var facetObj = _getStableFacets(scope);
 
+                    var isDescriptionMarkdown = columnModels.filter(function (model) {
+                        return model.column.name == "description"
+                    })[0].inputType == "longtext"
+
+                    function facetOptionsToString (options) {
+                        var str = "";
+                        options.forEach(function (option, idx) {
+                            var name = option.displayname.value;
+                            if (name === null) {
+                                str += " _No value_";
+                            } else if (name === "<i>All records with value </i>") {
+                                str += " _All records with value_"
+                            } else if (name === '') {
+                                str += " _Empty_"
+                            } else {
+                                str += " " + name;
+                            }
+                            if (idx+1 != options.length) str += ", "
+                        });
+                        return str;
+                    }
+
+                    // thresholds for when to use a modified simpler syntax for the name
+                    var facetChoicesThreshold = 5; // default to 5 choices
+                    var facetTextLengthThreshold = 60; // default to 50 characters;
+                    var nameLengthThreshold = 200; // default to 200 characters
+
+                    var nameDescriptionPrefix = scope.vm.reference.table.name + " with";
+                    var facetNames = "";
+
+                    var name = nameDescriptionPrefix
+                    var description = nameDescriptionPrefix + ":\n";
+
+                    var modelsWFilters = scope.vm.facetModels.filter(function (fm, idx) {
+                        fm.displayname = scope.vm.reference.facetColumns[idx].displayname.value;
+                        fm.preferredMode = scope.vm.reference.facetColumns[idx].preferredMode;
+                        return (fm.appliedFilters.length > 0);
+                    });
+
+                    if (scope.vm.search) {
+                        name += " " + scope.vm.search;
```

Comment:
I think you need a separator between search and the rest.

Reply:
fixed